src/Interpreters/JoinedTables.cpp

```diff
@@ -313,7 +313,7 @@ std::shared_ptr<TableJoin> JoinedTables::makeTableJoin(const ASTSelectQuery & se
             }
             else if (auto storage_dict = std::dynamic_pointer_cast<StorageDictionary>(storage); storage_dict)
             {
-                table_join->setStorageJoin(storage_dict);
+                table_join->setRightStorageName(storage_dict->getDictionaryName());
```

Comment:
Maybe without adaptor we can directly set dictionary storage using `setStorageJoin` method. Then there will be no special `right_storage_name`.

Reply:
Yes, now it's better